### DISCUSSION TITLE
High fidelity post stall and spin using JSBSim

### DISCUSSION THREAD

**Discussion by seanmcleod** [JSBSim-Team/jsbsim#361] [General]
I'm interested in trying to demonstrate that JSBSim can generate a high fidelity post stall and spin simulation.

I came across the following MSc thesis by Austin Murch - [Aerodynamic Modeling of Post-Stall and Spin Dynamics of Large Transport Airplanes](https://repository.gatech.edu/server/api/core/bitstreams/f54a68ca-4239-484b-ae75-1eaade7f4e9d/content) which describes his modelling of post stall and spin behavior based on extensive wind tunnel data from NASA.

Given all the data he used came from a NASA program there is a good chance that all the wind tunnel data is publicly available.

The wind tunnel data included the following:

- Static testing with large alpha and beta ranges
- Forced oscillation testing
- Rotary balance testing

![image](https://user-images.githubusercontent.com/824856/102016836-ad1bb000-3d6b-11eb-9aa9-5636a632fbab.png)

![image](https://user-images.githubusercontent.com/824856/102016824-9e34fd80-3d6b-11eb-9ffd-4ecf9df27f79.png)

He then also describes a couple of different methods that are typically used to blend the different wind tunnel data together during simulation and reports on his results.

In addition there is some "ground truth" data available from free spin tests that were done in the NASA vertical spin tunnel.

![image](https://user-images.githubusercontent.com/824856/102016809-7f366b80-3d6b-11eb-9ae6-c36d271e175d.png)

So we would be able to compare our JSBSim results against both Austin's simulation results and the data from the free spin tests.

Anyone interested in working on this?

**Comment by bcoconni** [comment]
Very nice ! That would bring a great improvement to JSBSim models.

I skimmed through the thesis (mainly looked at the pictures and read a small bit of text here and there 😄) The match is remarkable between predictions and - what I understood to be - wind tunnel results. It seems a very good place to start.

That is certainly an interesting topic to work on, however I am currently hacking JSBSim to address the issue #227 (I will open a separate discussion as soon as I have a working proof of concept - should be very soon). Anyway, I am interested in participating in this discussion: provide support, suggest ideas - if I can - and generally see how this is progressing.

**Comment by dany93** [comment]
I've added a stall and spin feature, up to flat spin, forward slipping, in the c172p FDM from the main FGAddon (fgdata pack). Initially experimental, but still alive since 2014.
Development version in [github c172p-team](https://github.com/c172p-team/c172p)
Not 'High Fidelity', even less based on a thesis, merely based on my basic understanding of the aerodynamics phenomena, as much as I could (I've never worked in the aircraft aerodynamics domain, only some experience on ultralight aircraft in real life with instructors). Unfortunately, I could not avoid using a few artifacts.
This stall and spin feature has been implemented on the P51D by Hal Van Engel too (we miss him).
Initially, I was working alone but, later, I learned that Hal Van Engel was working on this same subject with the same ideas, Then we communicated in this thread [trying to spin](https://forum.flightgear.org/viewtopic.php?f=25&t=21664&start=15#p197375).
Also, [Spin and Spin Recovery](https://forum.flightgear.org/viewtopic.php?f=25&t=7451#p71444).

**Comment by seanmcleod70** [comment]
[Deepak Paul Tirkey](https://www.linkedin.com/in/deepak-paul-tirkey-66b7b952/) recently posted on LinkedIn regarding the work he did to model spin behavior, coded up in Octave, using the MSc thesis by Austin Murch (mentioned above) as one of his references.

- [post 1](https://www.linkedin.com/feed/update/urn:li:activity:7478807738521686017/) 
- [post 2](https://www.linkedin.com/feed/update/urn:li:activity:7479189265474805760/)
